### ISSUE TITLE
void-variable counsel-find-file-map

### ISSUE THREAD

**Issue by mpanarin** [syl20bnr/spacemacs#12894]
#### Description :octocat:
After commit `383da46e38712ef6d08ef3e1e96d418bdf2182db` I started to get an error
`File mode specification error: (void-variable counsel-find-file-map)`
I think because I do not use counsel as well as it is not loaded in my configuration
Ivy layer is not installed

#### Reproduction guide :beetle:
- Remove Ivy layer
- Start Emacs
- SPC f e d

*Observed behaviour:* :eyes: :broken_heart:
You get an error message and completion or going to definition won't work

*Expected behaviour:* :heart: :smile:
Everything works

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.0.50
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 195ba3357)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(csv debug html yaml markdown docker helm ranger prodigy better-defaults emacs-lisp git lsp dap
     (python :variables python-fill-column 80 python-indent-offset 4 python-backend 'lsp)
     django
     (elixir :variables elixir-backend 'lsp)
     phoenix erlang sql tern
     (javascript :variables javascript-backend 'lsp javascript-fmt-tool 'web-beautify)
     react
     (typescript :variables typescript-backend 'lsp)
     rust themes-megapack org
     (shell :variables shell-default-height 30 shell-default-position 'bottom shell-default-shell 'vterm)
     spell-checking syntax-checking version-control
     (auto-completion :variables auto-completion-use-company-box t auto-completion-enable-sort-by-usage t auto-completion-enable-snippets-in-popup t)
     (treemacs :variables treemacs-use-follow-mode t treemacs-use-filewatch-mode t)
     (ibuffer :variables ibuffer-group-buffers-by 'projects)
     pdf games selectric xclipboard systemd restclient)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG CAIRO SOUND GPM DBUS GLIB NOTIFY INOTIFY ACL GNUTLS LIBXML2 FREETYPE HARFBUZZ LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD JSON PDUMPER LCMS2 GMP

**Comment by mpanarin** [comment]
added in this PR https://github.com/syl20bnr/spacemacs/pull/12866

I think because it was moved from ivy layer to a more generic one

**Comment by duianto** [comment]
I'm not getting an error message when pressing `SPC f e d` with the `helm` layer.

I even tried moving out the `ivy`, `counsel` and `swiper` packages from the `~.emacs.d/elpa/26.3/develop` directory.

`counsel-find-file-map` is only used:
```elisp
    (define-key counsel-find-file-map (kbd "C-h") 'counsel-up-directory)
```
in the function:
https://github.com/syl20bnr/spacemacs/blob/195ba335785fc068fa000756fab98bfe3813b3de/layers/%2Bspacemacs/spacemacs-completion/funcs.el#L192

Which is added to the hook:
https://github.com/syl20bnr/spacemacs/blob/195ba335785fc068fa000756fab98bfe3813b3de/layers/%2Bspacemacs/spacemacs-completion/packages.el#L132

in the function:
https://github.com/syl20bnr/spacemacs/blob/195ba335785fc068fa000756fab98bfe3813b3de/layers/%2Bspacemacs/spacemacs-completion/packages.el#L125

who's contents are wrapped in:
```elisp
  (with-eval-after-load 'ivy
```

>You get an error message and completion or going to definition won't work

`and completion`
could you elaborate with some reproduction steps?

I tried adding a message before the `counsel-find-file-map` key binding definition. But it's not being called when I open `.spacemacs` `SPC f e d` or when jumping to a definition `gd`.

**Comment by mpanarin** [comment]
@duianto Okay, so I was able to reproduce the issue more reliably on a blank linux and spacemacs install.

1. On the blank install, leave only `emacs-lisp`, `helm` and `treemacs` layers (vim editing style).
2. install ivy through `package-install` (in my particular case, a package `lispy` was installing swiper and ivy. But it didn't install counsel)
3. try to require ivy -> get error on counsel-mode-map.

the issue with definitions and completion not working was due to error above, and the rest of config not getting evaluated, so it doesn't really matter

**Comment by duianto** [comment]
Confirmed.

I opened a PR with a possible fix:
Fix void counsel-find-file-map #12898

**Comment by kirillsalykin** [comment]
Got same issue

#### System Info :computer:
- OS: darwin
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 195ba3357)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(auto-completion better-defaults
                 (clojure :variables clojure-enable-linters 'clj-kondo)
                 emacs-lisp git helm html markdown multiple-cursors osx sql syntax-checking treemacs version-control yaml)
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS

https://github.com/syl20bnr/spacemacs/pull/12898 fixes it for me (didnt notice any other errors)

**Comment by duianto** [comment]
The following PRs (possibly modified) changes are now on the develop branch:
Fix void counsel-find-file-map #12898
